### PR TITLE
add documentation for the consent api event to listen for

### PR DESCRIPTION
**Pull request by jazzsequence**
We need to be responsible for providing this documentation if we expect people to not dig down into the js api for this information.

This adds some information about the `wp_listen_for_consent_change` event that developers can listen to for a consent change, and copies the example code provided by the original consent api.

### CODE REVIEW

File: docs/developer-docs/JavaScript-Reference.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+
+Event to listen for consent changes.
+
+### Example
+
+```js
+document.addEventListener( 'wp_listen_for_consent_change', function ( e ) {
+  var changedConsentCategory = e.detail;
+  for ( var key in changedConsentCategory ) {
+    if ( changedConsentCategory.hasOwnProperty( key ) ) {
```
Comment (roborourke) [inline docs/developer-docs/JavaScript-Reference.md:11]: This example seems quite complex and a bit tricky to read. Wouldn’t `if ( e.detail.marketing && e.detail.marketing === 'allow' )` achieve the same?

**Review by roborourke — APPROVED** [review]
Nice addition. The example seems a bit complicated to me for what it’s doing, like I could see that code being useful if you wanted to iterate over every possible change but for illustrative purposes I think it could be a lot easier to understand the shape of the event detail object that you get back

File: docs/developer-docs/JavaScript-Reference.md (hunk 1 above)
Comment (jazzsequence) [reply docs/developer-docs/JavaScript-Reference.md:11]: probably, again this was copy pasta, but can update ours and maybe propose the change upstream as well.

```suggestion
    if ( e.detail.marketing && e.detail.marketing === 'allow' ) {
```

Comment (roborourke) [reply docs/developer-docs/JavaScript-Reference.md]: You could remove the `for` loop too with this change

**Comment by jazzsequence** [comment]
I really just wanted  to have _something_ relating to the event listener. I'm planning on demoing this today on the Americas hangout and realized this morning that we're a little light on _implementation_ docs that aren't just around "has consent".

At some point, I can dig a bit deeper and look at the properties of the event detail and we can add those to the docs.

**Pull request merged**